ethers-contract/ethers-contract-abigen/src/contract.rs

```diff
@@ -79,15 +79,14 @@ impl Context {
             #[allow(clippy::too_many_arguments)]
             mod #name_mod {
                 #imports
-
                 #struct_decl
 
-                impl<'a, M: Middleware> #name<M> {
+                impl<'a, M: ethers_providers::Middleware> #name<M> {
```

Comment:
Having fully qualified syntax here requires that we have to declare ethers_providers/core/contract in the consuming crate's cargo toml. It'd be ideal if we avoided that. Could you please revert this change?

Reply:
Instead of removing the fully qualified syntax, I simply renamed the top level imports to their crate name, which achieves the same effect, plus has the benefit, that the `EthEvent` trait, which resides in `ethers_contract` is usable with `ethers_contract` as crate dependency.

Reply:
Awesome, thank you, merged.